src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java

```diff
@@ -1,14 +1,19 @@
 package frc.robot.hardware.rev.motors;
 
 import com.revrobotics.spark.SparkLowLevel;
+import frc.robot.hardware.mechanisms.MechanismSimulation;
 
 public class BrushedSparkMAXMotor extends SparkMaxMotor {
 
-	public BrushedSparkMAXMotor(String logPath, SparkMaxWrapper motor) {
-		super(logPath, motor);
+	public BrushedSparkMAXMotor(String logPath, SparkMaxWrapper motor, MechanismSimulation mechanismSimulation) {
+		super(logPath, motor, mechanismSimulation);
 		if (motor.getMotorType() != SparkLowLevel.MotorType.kBrushed) {
 			throw new IllegalArgumentException("inserted BrushlessSparkMAXMotor to BrushedSparkMAXMotor!");
 		}
 	}
 
+	public BrushedSparkMAXMotor(String logPath, SparkMaxWrapper motor) {
+		this(logPath, motor, null);
```

Comment:
are you sure null is the correct thing to insert? is there no default constructor?

Reply:
If you look in the constructor it finally gets to, you can see that the only use for the mechanism simulation is in the create Simulation function, and it only returns a not Optional.empty object when the mechanism simulation!=null, so there is nothing risky about inserting null here